src/gcp/devConnect.ts

```diff
@@ -0,0 +1,244 @@
+  authorizerCredential?: OAuthCredential;
+  appInstallationId?: string;
+  installationUri?: string;
+}
+
+type InstallationStage =
+  | "STAGE_UNSPECIFIED"
+  | "PENDING_CREATE_APP"
+  | "PENDING_USER_OAUTH"
+  | "PENDING_INSTALL_APP"
+  | "COMPLETE";
+
+export interface InstallationState {
+  stage: InstallationStage;
+  message: string;
+  actionUri: string;
+}
+
+export interface Connection {
+  name: string;
+  createTime?: string;
+  updateTime?: string;
+  deleteTime?: string;
+  labels?: {
+    [key: string]: string;
+  };
+  githubConfig?: GitHubConfig;
+  installationState: InstallationState;
+  disabled?: boolean;
+  reconciling?: boolean;
+  annotations?: {
+    [key: string]: string;
+  };
+  etag?: string;
+  uid?: string;
+}
+
+type ConnectionOutputOnlyFields =
+  | "createTime"
+  | "updateTime"
+  | "deleteTime"
+  | "installationState"
+  | "reconciling"
+  | "uid";
+
+export interface GitRepositoryLink {
+  name: string;
+  cloneUri: string;
+  createTime: string;
+  updateTime: string;
+  deleteTime: string;
+  labels?: {
+    [key: string]: string;
+  };
+  etag?: string;
+  reconciling: boolean;
+  annotations?: {
+    [key: string]: string;
+  };
+  uid: string;
+}
+
+type GitRepositoryLinkOutputOnlyFields =
+  | "createTime"
+  | "updateTime"
+  | "deleteTime"
+  | "reconciling"
+  | "uid";
+
+export interface LinkableGitRepositories {
+  repositories: LinkableGitRepository[];
+  nextPageToken: string;
+}
+
+export interface LinkableGitRepository {
+  cloneUri: string;
+}
+
+/**
+ * Creates a Developer Connect Connection.
+ */
+export async function createConnection(
+  projectId: string,
+  location: string,
+  connectionId: string,
+  githubConfig: GitHubConfig,
+): Promise<Operation> {
+  const config: GitHubConfig = {
+    ...githubConfig,
+    githubApp: "FIREBASE",
+  };
+  const res = await client.post<
+    Omit<Omit<Connection, "name">, ConnectionOutputOnlyFields>,
+    Operation
+  >(
+    `projects/${projectId}/locations/${location}/connections`,
+    {
+      githubConfig: config,
+    },
+    { queryParams: { connectionId } },
+  );
+  return res.body;
+}
+
+/**
+ * Gets details of a single Developer Connect Connection.
+ */
+export async function getConnection(
+  projectId: string,
+  location: string,
+  connectionId: string,
+): Promise<Connection> {
+  const name = `projects/${projectId}/locations/${location}/connections/${connectionId}`;
+  const res = await client.get<Connection>(name);
+  return res.body;
+}
+
+/**
+ * List Developer Connect Connections
+ */
+export async function listConnections(projectId: string, location: string): Promise<Connection[]> {
+  const conns: Connection[] = [];
+  const getNextPage = async (pageToken = ""): Promise<void> => {
+    const res = await client.get<{
+      connections: Connection[];
+      nextPageToken?: string;
+    }>(`/projects/${projectId}/locations/${location}/connections`, {
+      queryParams: {
+        pageSize: PAGE_SIZE_MAX,
```

Comment:
this seems at odds with `fetchLinkableGitRespositories()` below, is PAGE_SIZE_MAX intended to be the max value or the default value?

Reply:
Maybe we should have a listConnections vs listAllConnections?